noir-projects/aztec-nr/aztec/src/context/private_context.nr

```diff
@@ -379,49 +375,25 @@ impl PrivateContext {
         //     self.min_revertible_side_effect_counter = item.public_inputs.min_revertible_side_effect_counter;
         // }
 
-        assert(contract_address.eq(item.contract_address));
-        assert(function_selector.eq(item.function_data.selector));
-
-        assert(args_hash == item.public_inputs.args_hash);
-
-        // Assert that the call context of the call generated by the oracle matches our request.
-        assert(item.public_inputs.call_context.is_delegate_call == is_delegate_call);
-        assert(item.public_inputs.call_context.is_static_call == is_static_call);
-
-        if (is_delegate_call) {
```

Comment:
The same checks for (non_)delegate call is done in the kernel and must be done there. It's not necessary to check them here in the app circuit. If we want to abort early before executing kernel circuits, can check and throw from the simulator.